{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genomic Data Visualization\n",
    "- Circos: circular aggrangement of the genome \n",
    "    - Whole genome overview\n",
    "    - Hard to compare\n",
    "- Sequences: UCSC Genome Browser\n",
    "- Sequences: IGV (Integrative Genomics Viewer)\n",
    "    - not a browser, they said\n",
    "    - zoom in to one thing, but that makes loosing a lot of context\n",
    "    \n",
    "## Taxonomies for Genomic Data Visualization\n",
    "### Challenges\n",
    "- Genome Organization\n",
    "    - Multiple chromosomes\n",
    "    - Stored as linear sequence, but interations are non-linear:short-range, long range\n",
    "- Pattern across Scales\n",
    "    - Chromosome level to individual nucleotides(A,C,G,T)\n",
    "- Many types of patterns along the genome(SNP, epigenomic peaks, genome ic rearragements, etc.)\n",
    "- Everything is conected\n",
    "    - Need to integrate different data types: sequence, expresssion levels, metabolites, phenotype information, etc\n",
    "    - Need to load many different types of data into a sigle software\n",
    "- Large space with sparse distribution of patterns across multiple scales\n",
    "- Large data sets(in bytes)\n",
    "- Privacy concerns(data access)\n",
    "\n",
    "#### Gremlin\n",
    "- 여러 scale을 한꺼번에 보여주면서 scale issue를 해결할 수 있다.\n",
    "- sample data이고, 공개가 되지 않음.\n",
    "\n",
    "#### [HiGlass](https://higlass.io/)\n",
    "- scale을 zoom in 하면서 detail을 볼 수 있다. \n",
    "- multiviews가능하다. \n",
    "- space / scale 문제를 어느정도 해결\n",
    "\n",
    "### Approach\n",
    "- Only consider data that is visualized in the sequence context\n",
    "- Treat genome as a coordinate system\n",
    "- Seperate concerns:\n",
    "    - What data types can be mapped to the genome?\n",
    "    - How can the coordinate system be laid out?\n",
    "    - How can the mapped data be encoded?\n",
    "    - How can the mapped data be encoded?\n",
    "    - What kind of tasks are users trying to address with genomic data?\n",
    "\n",
    "#### Data Taxonomy\n",
    "- Feature Sets\n",
    "\n",
    "#### Visual Taxonomy\n",
    "- Coordinate System\n",
    "    - Layout\n",
    "        - linear\n",
    "        - circular\n",
    "        - space-filling\n",
    "        - spatial(3-Dimensional) e.g. folding\n",
    "    - Partition\n",
    "        - Contiguous: end to end\n",
    "        - Segregated: nextto each other, not connected\n",
    "    - Abstraction\n",
    "        - None\n",
    "        - Partial e.g. only shows exon\n",
    "        - Complete e.g. length of the exome, only gene basepair\n",
    "    - Arrangement\n",
    "        - Linear\n",
    "            - Parallel\n",
    "            - Serial\n",
    "            - Orthogonal e.g. conservation\n",
    "        - Circular\n",
    "            - Parallel\n",
    "            - Serial    \n",
    "    - Tracks : 색깔은 많아서 재미있을지는 몰라도 특별한 것은 없다. \n",
    "        - Encoding\n",
    "        - Argument\n",
    "    - View configurations\n",
    "        - Views: 한꺼번에 얼마나 많은 view를 볼 수 있는가? \n",
    "        - Scales: How many different scale you can look at a time point?\n",
    "        - Foci: How many different loci you can look at a time point?\n",
    "    \n",
    "#### Task Taxonomy\n",
    "\n",
    "#### Examples of the Genetic Visualizations\n",
    "##### Over view\n",
    "* e.g.[http://epiviz.cbcb.umd.edu/4/](http://epiviz.cbcb.umd.edu/4/): single view/single scale/single foci  \n",
    "* e.g.[http://civi.cmbi.ru.nl](http://civi.cmbi.ru.nl) : single view / single scale/ sinlge foci  \n",
    "* e.g. gremlin: multiple view / multiple scale / 1 focus view  \n",
    "* e.g. vials gene: single giew / single scale / single foci  \n",
    "* e.g. software.broadinstitute.org/software/igv/  \n",
    "* e.g. Higplass: multiple view / multiple scale / multiple focus view  \n",
    "* e.g. [Rondo](http://odonoghuelab.org:8020): single view / single scale / single focus  \n",
    "* e.g. [Givengine](https://mcf7.givengine.org)\n",
    "* e.g. [Sequencelogo](http://weblogo.berkeley.edu/logo.cgi)\n",
    "* e.g. [Vistapoint](http://genome.lbl.gov/vista/index.shtml)\n",
    "* e.g. [cinteny.cchmc.org](http://cinteny.cchmc.org) - Too many color - it is not good solution\n",
    "* e.g. [Mizbee](http://www.cs.utah.edu/~miriah/mizbee/Overview.html)\n",
    "![](http://www.cs.utah.edu/~miriah/mizbee/Overview_files/fish_overview.jpg)\n",
    "* e.g. [SynMap](https://genomevolution.org/CoGe/SynMap.pl): 다른 동물간에 chromosome 비교\n",
    "\n",
    "##### smaller region\n",
    "* e.g. Lollipop figure: [cbioportal](http://cbioportal.org) \n",
    "\n",
    "#### View를 생각할 것\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "참고: [http://genocat.tools/](http://genocat.tools/)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# State of the Art Reports(STARTs)\n",
    "## Garphs\n",
    "\n",
    "## Tubular data\n",
    "\n",
    "## Sets\n",
    "\n",
    "## Imaging\n",
    "\n",
    "## Molecular Structures\n",
    "\n",
    "## General \n",
    "- integrating machine lerning into visual analytics (2017)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Exmples"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "* tree visualization: [http://treevis.net](http://treevis.net)\n",
    "* time visualization: [http://timevis.net](http://timevis.net)\n",
    "* set visualization\n",
    "* dynamic visualization\n",
    "* [sos.swansea.ac.uk](http://sos.swansea.ac.uk): survey"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
